VirtualReality/MRML/vtkVirtualRealityViewInteractorObserver.cxx

```diff
@@ -0,0 +1,294 @@
+    }
+}
+
+//----------------------------------------------------------------------------
+void vtkVirtualRealityViewInteractorObserver::CustomProcessEvents(vtkObject* object,
+  unsigned long event, void* clientdata, void* calldata)
+{
+  vtkVirtualRealityViewInteractorObserver* self
+    = reinterpret_cast<vtkVirtualRealityViewInteractorObserver *>(clientdata);
+
+  if (!self->DelegateInteractionEventToDisplayableManagers(event) || self->GetInteractorStyle()->GetState() != VTKIS_NONE)
+    {
+    // Displayable managers did not processed it
+    vtkVirtualRealityViewInteractorObserver::ProcessEvents(object, event, clientdata, calldata);
+    }
+}
+
+//----------------------------------------------------------------------------
+void vtkVirtualRealityViewInteractorObserver::ProcessEvents(
+  vtkObject* object, unsigned long event, void* clientdata, void* calldata)
+{
+  Superclass::ProcessEvents(object, event, clientdata, calldata);
+
+  vtkVirtualRealityViewInteractorObserver* self =
+    reinterpret_cast<vtkVirtualRealityViewInteractorObserver*>(clientdata);
+
+  switch (event)
+    {
+
+    /// 3D event bindings
+    // Move3DEvent: Already observed in vtkMRMLViewInteractorStyle
+    // Button3DEvent: Already observed in vtkMRMLViewInteractorStyle
+    case vtkCommand::Menu3DEvent:
+      self->OnMenu3D(static_cast<vtkEventData*>(calldata));
+      break;
+    case vtkCommand::Select3DEvent:
+      self->OnSelect3D(static_cast<vtkEventData*>(calldata));
+      break;
+    case vtkCommand::NextPose3DEvent:
+      self->OnNextPose3D(static_cast<vtkEventData*>(calldata));
+      break;
+    case vtkCommand::ViewerMovement3DEvent:
+      self->OnViewerMovement3D(static_cast<vtkEventData*>(calldata));
+      break;
+    case vtkCommand::Pick3DEvent:
+      self->OnPick3D(static_cast<vtkEventData*>(calldata));
+      break;
+    case vtkCommand::PositionProp3DEvent:
+      self->OnPositionProp3D(static_cast<vtkEventData*>(calldata));
+      break;
+    case vtkCommand::Clip3DEvent:
+      self->OnClip3D(static_cast<vtkEventData*>(calldata));
+      break;
+    case vtkCommand::Elevation3DEvent:
+      self->OnElevation3D(static_cast<vtkEventData*>(calldata));
+      break;
+
+    default:
+      break;
+    }
+}
+
+
+//----------------------------------------------------------------------------
+// Generic events binding
+//----------------------------------------------------------------------------
+
+//----------------------------------------------------------------------------
+bool vtkVirtualRealityViewInteractorObserver::DelegateInteractionEventToDisplayableManagers(vtkEventData* inputEventData)
+{
+  // Get display and world position
+  int* displayPositionInt = this->GetInteractor()->GetEventPosition();
+  vtkRenderer* pokedRenderer = this->GetInteractor()->FindPokedRenderer(displayPositionInt[0], displayPositionInt[1]);
+  if (!pokedRenderer || !inputEventData)
+  {
+    // can happen during application shutdown
+    return false;
+  }
```

Comment:
Add this to Slicer in `vtkMRMLViewInteractorStyle::DelegateInteractionEventToDisplayableManagers`